types/collection.hh

```diff
@@ -54,6 +54,8 @@ public:
     virtual bool is_value_compatible_with_frozen(const collection_type_impl& previous) const = 0;
     template <typename BytesViewIterator>
     static bytes pack(BytesViewIterator start, BytesViewIterator finish, int elements, cql_serialization_format sf);
+    template <std::forward_iterator BytesViewIterator>
+    static managed_bytes pack_fragmented(BytesViewIterator start, BytesViewIterator finish, int elements, cql_serialization_format sf);
```

Comment:
Please add a constraint on the value_type of the iterator.

Reply:
This function is later replaced with a version which takes fragmented inputs, and that final version has the constraint, but I can add it to this commit too.

Reply:
I see that in a later patch (6756eaa9df4829c8ad9ec4f6835e1f5cbee62258) even the std::forward_iterator is constraint is removed. Confused.

Reply:
Oops, the change in 6756eaa doesn't do anything. It shouldn't be there, sorry for the confusion. e1ce9b0a add the constraints.

Reply:
Fixed.